server/src/main/java/org/opensearch/wlm/ResourceType.java

```diff
@@ -61,20 +75,30 @@ public String getName() {
         return name;
     }
 
-    /**
-     * Gets the resource usage for a given resource type and task.
-     *
-     * @param task the task for which to calculate resource usage
-     * @return the resource usage
-     */
-    public long getResourceUsage(Task task) {
-        return getResourceUsage.apply(task);
-    }
-
     public boolean hasStatsEnabled() {
         return statsEnabled;
     }
 
+    public double calculateQueryGroupUsage(List<QueryGroupTask> tasks, Supplier<Long> nanoTimeSupplier) {
+        return resourceUsageCalculator.calculateResourceUsage(tasks, nanoTimeSupplier);
+    }
+
+    public double calculateTaskUsage(QueryGroupTask task, Supplier<Long> nanoTimeSupplier) {
+        return resourceUsageCalculator.calculateTaskResourceUsage(task, nanoTimeSupplier);
+    }
+
+    public boolean isBreachingThreshold(QueryGroup queryGroup, double currentUsage) {
+        return getExcessUsage(queryGroup, currentUsage) > 0;
+    }
+
+    public double getExcessUsage(QueryGroup queryGroup, double currentUsage) {
+        return resourceUsageUtil.getExcessUsage(queryGroup, currentUsage);
+    }
```

Comment:
I am wondering if we should really write these wrappers on top of ResourceUsageUtil. Probably better to just expose the underlying object?

Reply:
These wrapper will work as adapter to any changes in underlying member variables.
Hence I think this is more appropriate.